src/sentry/api/utils.py

```diff
@@ -111,3 +113,26 @@ def is_member_disabled_from_limit(request, organization):
         return False
     else:
         return member.flags["member-limit:restricted"]
+
+
+def generate_customer_hostname(org_slug: str) -> str:
+    url_prefix_hostname = urlparse(options.get("system.url-prefix")).netloc
+    customer_base_hostname_template = options.get("system.customer-base-hostname")
+    if not customer_base_hostname_template:
+        return url_prefix_hostname
+    if "{slug}" not in customer_base_hostname_template:
```

Comment:
Could you add unit tests that test this logic (and the other conditionals below)